include/spineml/simulator/inputValue.h

```diff
@@ -166,14 +167,35 @@ class External : public Base
     //------------------------------------------------------------------------
     // InputValue virtuals
     //------------------------------------------------------------------------
-    virtual void update(double dt, unsigned int timestep,
-                        std::function<void(unsigned int, double)> applyValueFunc) override;
+    virtual void update(double dt, unsigned long long timestep,
+                        std::function<void(unsigned int, double)> applyValueFunc) final;
+
+    //------------------------------------------------------------------------
+    // Public API
+    //------------------------------------------------------------------------
+    std::vector<double>::iterator getBufferBegin(){ return m_Buffer.begin(); }
+    std::vector<double>::iterator getBufferEnd(){ return m_Buffer.end(); }
```

Comment:
Well this is an input so you can't do much to it with a const pointer. However I didn't want to expose a non-const reference to ``m_Buffer`` as that would let you fuck it up in many ways that would break the internals.

Reply:
That's a good point.